packages/ui-toolkit/src/components/atoms/NumberInput/NumberInputStepper.tsx

```diff
@@ -56,7 +61,8 @@ const StepPrefixComponent = ({ step = 1, min = Number.NEGATIVE_INFINITY, value,
 
       // @ts-ignore : to prevent onChange re writing as it can be passed by user
       //we are synthentically generating custome event to set value
-      onChange({ target: { value: floorValue } });
+      // sending actionType to detect that value has been decreased by clicking on the - icon
+      onChange({ target: { value: floorValue }, actionType: 'DECREMENT' });
```

Comment:
Why this is needed? Doesn't seem like a need for this API. Can we handle this in the usage level instead of doing it here?

Reply:
Discussed 👍